perun-core/src/main/java/cz/metacentrum/perun/core/impl/ExtSourceREMS.java

```diff
@@ -0,0 +1,157 @@
+		return filterNonExistingUsers(subjects);
+	}
+
+	@Override
+	public List<Map<String, String>> findSubjects(String searchString, int maxResults) throws InternalErrorException, ExtSourceUnsupportedOperationException {
+		List<Map<String, String>> subjects = super.findSubjects(searchString, maxResults);
+		return filterNonExistingUsers(subjects);
+	}
+
+	@Override
+	public Map<String, String> getSubjectByLogin(String login) throws InternalErrorException, SubjectNotExistsException {
+		Map<String, String> subject = super.getSubjectByLogin(login);
+		if (!isExistingUser(subject)) {
+			throw new SubjectNotExistsException("Subject for given login does not exist in Perun");
+		}
+		return subject;
+	}
+
+	@Override
+	public List<Map<String, String>> findSubjectsLogins(String searchString) throws InternalErrorException {
+		List<Map<String, String>> subjects = super.findSubjectsLogins(searchString);
+		return filterNonExistingUsers(subjects);
+	}
+
+	@Override
+	public List<Map<String, String>> findSubjectsLogins(String searchString, int maxResults) throws InternalErrorException {
+		List<Map<String, String>> subjects = super.findSubjectsLogins(searchString, maxResults);
+		return filterNonExistingUsers(subjects);
+	}
+
+	@Override
+	public List<Map<String, String>> getGroupSubjects(Map<String, String> attributes) throws InternalErrorException {
+		List<Map<String, String>> subjects = super.getGroupSubjects(attributes);
+		return filterNonExistingUsers(subjects);
+	}
+
+	/**
+	 * Filters subjects that does not have a corresponding user in Perun by ues REMS
+	 * or by additionalues_1 in format: {extSourceName}|{extSourceClass}|{eppn}|0
+	 *
+	 * @param subjects subjects that will be filtered
+	 * @return List without filtered subjects
+	 * @throws InternalErrorException
+	 */
+	private List<Map<String, String>> filterNonExistingUsers(List<Map<String, String>> subjects) throws InternalErrorException {
+		List<Map<String, String>> existingSubjects = new ArrayList<>();
+
+		for (Map<String, String> subject : subjects) {
+			if(isExistingUser(subject)) {
+				existingSubjects.add(subject);
+			}
+		}
+
+		return existingSubjects;
+	}
+
+	/**
+	 * Checks if the given subject has a corresponding user in Perun by ues REMS
+	 * or by additionalues_1 in format: {extSourceName}|{extSourceClass}|{eppn}|0
+	 *
+	 * @param subject subject
+	 * @return true if the subject has, false otherwise
+	 * @throws InternalErrorException
+	 */
+	private boolean isExistingUser(Map<String, String> subject) throws InternalErrorException {
+		if (subject.isEmpty()) {
+			return false;
+		}
+		String login = subject.get("login");
+
+		PerunSession sess = getSession();
+
+		List<User> usersFromREMS = perunBl.getUsersManagerBl().getUsersByExtSourceTypeAndLogin(sess, ExtSourcesManager.EXTSOURCE_REMS, login);
+		if (usersFromREMS.size() > 0) {
+			return true;
+		}
+
+		String extSource = subject.get("additionalues_1");
```

Comment:
This seems to be very strict setting. You should try all "additionalues_x" and if there is none, you can log it and return false instead of throwing and exception. What do you think?